src/v/config/configuration.cc

```diff
@@ -752,7 +754,43 @@ configuration::configuration()
       "Default replication factor for new topics",
       {.needs_restart = needs_restart::no, .visibility = visibility::user},
       1,
-      {.min = 1, .oddeven = odd_even_constraint::odd})
+      {.min = 1,
+       .oddeven = odd_even_constraint::odd,
+       .additional_validation =
+         [this](const int16_t& default_value) -> std::optional<ss::sstring> {
+           auto& minimum_val = minimum_topic_replication_factor.value();
+           if (minimum_val.has_value() && *minimum_val > default_value) {
+               return ss::format(
+                 "Cannot set '{}' to a value less than "
+                 "'{}' ({})",
+                 default_topic_replication.name(),
+                 minimum_topic_replication_factor.name(),
+                 *minimum_val);
+           }
+           return std::nullopt;
+       }})
+  , minimum_topic_replication_factor(
+      *this,
+      "minimum_topic_replication_factor",
+      "Minimum value permitted for a topic's replication factor",
+      {.needs_restart = needs_restart::no, .visibility = visibility::user},
+      std::nullopt,
+      {.min = 3,
```

Comment:
This means that single node setups (like our docker container via `rpk container start`) will not be able to create topics?

Reply:
By default this config is disabled so no, it won't prevent normal operations in a single-node cluster.

Reply:
From reading the cover letter seems like by default this is not set so it has no effect, and even if it was it would only generate a WARN, is this correct?

Reply:
> They both need to default to 1, we'll just have to override this in cloud

I disagree here.  The purpose of this config is to prevent users of Redpanda from creating topics with an RF of 1.  This would be used in multi-node clusters to enforce the minimum to protect users data.

The default value comes in when RF is not supplied when a topic is created.

> From reading the cover letter seems like by default this is not set so it has no effect, and even if it was it would only generate a WARN, is this correct?

The WARN message only occurs for topics created _before_ the minimum value is specified.

Reply:
Ah I missed that it's a `std::optional`. LGTM then